ethcore/src/blockchain/blockchain.rs

```diff
@@ -1093,6 +1148,20 @@ impl BlockChain {
 		}
 	}
 
+	/// Iterator that lists `first` and then all of `first`'s ancestors, by extended header.
+	pub fn ancestry_header_iter<'a>(&'a self, first: H256) -> Box<Iterator<Item=ExtendedHeader> + 'a> {
```

Comment:
the `ancestry_iter` method also returns headers. we could call this `ancestry_with_metadata`